keps/sig-node/20191118-cgroups-v2.md

```diff
@@ -0,0 +1,229 @@
+---
+title: cgroups v2
+authors:
+  - "@giuseppe"
+owning-sig: sig-node
+participating-sigs:
+      - sig-architecture
+reviewers:
+  - "@yujuhong"
+  - "@dchen1107"
+  - "@derekwaynecarr"
+approvers:
+  - "@yujuhong"
+  - "@dchen1107"
+  - "@derekwaynecarr"
+editor: Giuseppe Scrivano
+creation-date: 2019-11-18
+last-updated: 2019-11-18
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Cgroups v2
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Goals](#goals)
+- [User Stories](#user-stories)
+- [Implementation Details](#implementation-details)
+  - [Dependencies on OCI and container runtimes](#implementation-details)
+  - [Current status of dependencies](#current-status-of-dependencies)
+  - [Current cgroups usage and the equivalent in cgroups v2](#current-cgroups-usage-and-the-equivalent-in-cgroups-v2)
+- [Proposal](#proposal)
+  - [Option 1: Convert from cgroups v1 settings to v2](#option-1-convert-from-cgroups-v1-settings-to-v2)
+  - [Option 2: Use cgroups v2 throughout the stack](#option-2-use-cgroups-v2-throughout-the-stack)
+- [Risk and Mitigations](#risk-and-mitigations)
+- [Graduation Criteria](#graduation-criteria)
+- [Testing](#testing)
+
+<!-- /toc -->
+
+## Summary
+
+A proposal to add support for cgroups v2 to kubernetes.
+
+## Motivation
+
+The new kernel cgroups v2 API was declared stable more than two years
+ago. Newer features in the kernel such as PSI depend upon cgroups
+v2. groups v1 will eventually become obsolete in favor of cgroups v2.
+Some distros are already using cgroups v2 by default, and that
+prevents Kubernetes from working as it is required to run with cgroups
+v1.
+
+## Goals
```

Comment:
May I suggest to explicitly document non-goal at the beginning of doc too for a better communication? For example, podspec changes to enable some advanced features, dockershim, etc.

Reply:
sure, I'll add it